### ISSUE TITLE
Even more project involvement!

### ISSUE THREAD

**Issue by tngreene** [X-Plane/XPlane2Blender#594]
#453 is a pretty sad bug report. Here's the core of what's left in it:

# Ease of contributing to the docs
- [ ] It would be useful (even for myself) to have a style guide explaining what gets what formatting!
- [ ] The big thing users could provide are tutorials and example files for people to download.
- [ ] Now's a good time to take new screenshots with 2.9, because the UI won't change much (finally! Turns out that not everyone can simply map 2.79->2.80 quickly.)

Above all, instead of waiting for me is a bad idea, so please, go make a tutorial on the forms and help people learn that way
- [ ] What's the best way someone could actually contribute to XPlane2Blender-docs? GitBooks is so touchy I'd rather people just send me e-mails than touch it.

# Ease of contributing to the code
- [ ] We have many bugs marked "Help Wanted". I think all I can do is to lean back and wait. Perhaps I can broadcast on the forms that there are easy ways to get involved in programming the "nice to haves" that people ask for. If anyone has been curious and knows anyone, have them come and pick a bug!

# Ease of getting more unit tests
- [ ] The unit test could be even easier. For "check the fixture that has the name of this function, for the following fixtures, that could be 1 line
- [ ] A tutorial on how it all links together would be useful. Its all pretty abstract.

**Comment by tngreene** [comment]
**And a massive thank you to everyone who steps in and helps each other out with the basic bugs like how to install something!** This has been, honestly, the best thing I could have possibly hoped for!

**Comment by kbrandwijk** [comment]
I'm not trying to bash the idea here, or be negative about it, but I would like to share my perspective on this, at least from a 'contributing to the code' point of view, as that's where my personal interest is.

Here's the main thing, the cycles have been pretty large. If a sizeable contribution sits around for 10 (!) months before getting anywhere close to being integrated, that doesn't work. That's not how opensourcing a tool works. And yes, I understand there's a certain roadmap, and Blender 2.8+ was a huge deal, and all that. But there's 0 satisfaction, and 0 incentive to continue contributing, if you feel like your contributions just end up on the shelf. Having open PRs from 1, 3 and even 4 years ago doesn't look inviting either.

You also suggest taking a look at the 'good first issue' or 'help wanted' issues, while you sit back and wait for us to do so (bit of a strange way with words maybe). I've been glancing over those, and I have a pretty solid understanding of the codebase. Some of them are close to 3 years old, the information (if any) in half of them is deprecated at best, and from a developer point of view, there's not many that I would consider your average 'welcome to the codebase' level issues. If I was to pick something up right now to work on, nothing jumps out to me as either extremely useful, or a quick win.

To be honest, you've closed #453 as being a 'sad story', but it was somehow more sincere than this new one...

**Comment by DWmFrancis** [comment]
I’m relatively new to the conversation, so I lack history and may come off like a noob/doof, but in terms of both presentation and obviousness, the user interface needs work. It looks like it was written by experts for experts and there are some UX errors which I’d fail one of my design students for.

One of the most compelling market differentiators for X-Plane was providing the tools to build your own aircraft. 
From what I’m seeing happening with FS2020, that edge may not be the case much longer.

If you are looking for something to do, take a look at that - and yes, I’ll offer concrete suggestions.

Those “sad stories” are legitimate calls for help and I guarantee that for every person who takes the time to write, there are 99 who have the same issue but don’t mention it.

-df

**Comment by tngreene** [comment]
@kbrandwijk and @DWmFrancis , it is very nice to hear from you again! Thank you for your long responses!

**Firstly let me say when I said "sad" I'm not talking about anyone's response being sad or making me sad, I'm talking how it makes **me** sad thinking about the last 10 months in my life and XPlane2Blender's long v4.0.0 release cycle.** #453 was written on a day when I was feeling overwhelmed but very hopeful - that somehow taking on a third major facet that involved other people would be boost to myself and the project. It sounds crazy but there is a saying "get busy to stay busy" and "its easier to be held accountable to others than to will yourself to be accountable to yourself." And then the next 10 months of my life happened and every time I've looked at that bug I see a big wasted opportunity - like wasting the opportunity to get to know you @kbrandwijk! That's a pretty big damn waste in my book - a big sad regret seeing energy and well stated user's concerns that went no where because I wasn't there to help grow it. But, I hope you'll excuse me if I don't elaborate about my personal health on the internet.

Now that things are better and I have completed several objectives I set out in it I wanted a place that could track further progress and plan out the future. That's what "sad" means and that's what this bug means - progress based on the comments written in #453 and now here.

@kbrandwijk, you bring up excellent points
- The release cycle was far too long, and I can see how disappointing that would be, Next release cycles will have 1 or 2 features or large bug fixes __at most__ and hopefully some kind of road map can be established that would be helpful. I'll add that above
- I can see how PR requests from 4 years ago would look bad. I don't know what is the best thing to do with them, A clean slate there would look more inviting
- I can see how so many "Good first issues" could make it hard to pick any of them to start. Applying them less frequently would probably be best. Right now it stands for "easy". Help wanted is kind of useless, because the only bugs I don't want help on the ones I'm working on. Which, I don't know if I express well
- "half are 3 years old and have deprecated information at best", I do look through the old bugs often to see if any can be closed or combined. I don't feel that so many have become too out of date to work on, especially many little features that haven't been coded. If you would like to give me some examples it would be extremely useful
- "not many I'd call "entry level"". You're right, I've over used the label. Do you have any suggestions about what is "entry level"? Obviously some documentation about the codebase itself would be useful to get people started.
- "To be honest, you've closed #453 as being a 'sad story', but it was somehow more sincere than this new one..." and "Those “sad stories” are legitimate calls for help and I guarantee that for every person who takes the time to write, there are 99 who have the same issue but don’t mention it."

This makes me so unhappy to hear. I'll say it again, and I'm re-writing the initial post to reflect this - When I posted#453 that I was shocked to see such well written thoughts being given. I really thought I was shouting into the void. Those responses mean so much to me! And then that dream didn't come through how I wanted and it was hard to make progress on it and keep it actionable and useful. So, I closed it and made this so I could stop looking at what I didn't accomplish and my high hopes that didn't come true. I'm still not sure where this goes, but, I know this fresh start has a chance, where as the previous is full of good lessons to refer to but not something that I can easily read and use.

**Comment by tngreene** [comment]
@DWmFrancis I agree that Blender's UI is limiting. I ready UI/UX blogs constantly and think "Oh, that pattern would make it so much easier to do ________, wait, Blender's UI framework is terrible. Why did I bother reading this." Its big trouble is that the UI is the data model, basically. The checkbox for "Cast Shadow (Local)" is the same variable as what gets used in the exporter. **You get very little control, very few options, and changing the UI means changing the exporter code itself.** We're in the same boat of wishing Blender could do more.

More UI means more properties and more properties means more trouble with updating. To remove a drop down menu to an int prop means updater code. Bad updater code of simply re-ordering a drop down menu nearly destroyed someone's work. Hence why I avoid updater code, or changing the UI (and remind all to make backups!)

It is truly pathetic in my opinion and Blender should have done better (ever heard of Model-View-Controller?!) If you wanted to research other Blender addons and how they use hacks to make a better UI than Blender is supposed to offer, it would be very useful.

Lastly, I've read every e-mail you've ever sent me (and hopefully gotten to respond to most of them.) When you said "Fusion 360" I immediately thought "I wonder how many other people try to use this in their workflow?" and reminded me that we need a tutorial on baking textures. Thank you for being the 1 that shows that the many silent are thinking!

**Comment by tngreene** [comment]
This bug represents a very long term goal, but, without defining a goal it goes nowhere. One day we'll get due dates on this stuff and incremental changes will build until we're really taking off and every artistically inclined user can download Blender and XPlane2Blender and get started faster than starting with World Editor!

(Yes, this is secretly all a competition with Michael Minnhaar for who has the best tools - Scenery and Planes vs Airports! 😛 )

**Comment by DWmFrancis** [comment]
Ted - 

One thing which occurred to me is that with Blender being so configurable - I think there’s a preference for *everything* in there somewhere - something which might simplify the UI would be to create a loadable x-Plane preference set which would limit the windows and tools  to the things X-Plane builders need most.

The other thing to reflect on is that the folks over in MSFS Land are also using Blender and posting some very interesting videos on YouTube about how to do it.  

Here is an example of one which directly addresses one of my major pain points; UV UnWrapping and Texture Painting;

https://youtu.be/SZCe_x-V9co <https://youtu.be/SZCe_x-V9co>

Note that they also have provided a way to import aircraft (MSFS gITF) files into Blender.


On the other hand - updating PlaneMaker to support much “rounder" fuselages that don’t look like the Hindenberg under all that paint and providing a way to punch holes into the walls that are instantly transparent plexiglas windows - or maybe drag and drop windows (stencils?/decals?) might be another way to improve the appearance of Nose Sections and Fuselages.

In other words; Don’t force users to become mechanics in order to drive their cars.
 
-df

**Comment by airfightergr** [comment]
@DWmFrancis I think you are a bit confused. Planemaker is a cad-like application to build the physics simulation. Blender and XPlane2Blender have nothing to do with planemaker, and there is no connection between. 

With in mind any last decade workflow, the planemaker stuff (3D geometry), is meant to be hidden and show only the 3D geometry you build in Blender. That's why no one cares how the planemaker geometry looks. You only need a geometry that is good enough to approximate aircraft's flight characteristics.

On the video you posted, the only merit I can see is if a livery painter want to use Substance Painter for the liveries, instead of Photoshop or Gimp, to have the 3D object available to paint on. Though there are many things to consider here. For example, a 3D object from any aircraft (even from Laminar's ones), are not open source, and might not be used in any form without proper license. 

Most developers are providing with some sort of paintikit for use in Photoshop/Gimp for those who want to do their own liveries. If someone want to use Substance painter and needs the 3d object to paint upon, as a developer, I might be happy to provide a blender file with the parts needed for doing so. Or even a file with the proper format for importing it in Substance painter.

**Comment by DWmFrancis** [comment]
Ilias - 

I’m not sure what you think I’m confused about. 

PlaneMaker creates a crude 3D model of the primary elements of the airframe (Fuselage, Wings, Horizontal and Vertical Stabilizers, Nacelles, etc. and the container for the data refs which define the aircraft performance. I’ve used it hundreds of times developing and editing aircraft for X-Plane. It also provides a way to create and populate the instrument panels.

I also understand that the current approach to developing 3D aircraft is to do the airframe and cockpit development external to Plane Maker and only use it to gather the needed data refs to accurately model the aircraft performance. 

My concern is the difficulty novice aircraft builders face and how that impacts the aircraft creation and customization segments of the X-Plane community. 
Bender and PlaneMaker are barriers to entry - Blender’s steep learning curve in particular. Another consideration is the huge fleet of legacy aircraft which were created for previous versions of X-Plane. 

It’s good of you to be happy to provide a Blender file of your aircraft, but I’m not sure all the developers will share your generous spirit.

So, my point to Ted (and Austin) is that one of PlaneMaker’s significant market differentiators was it’s well integrated  “drag and drop” nature. 
Over the years the value of that has degraded, but it’s an area that Laminar could still leverage that would build on it’s DIY roots.

-df

**Comment by tngreene** [comment]
@DWmFrancis I've never touched nor will I ever touch PlaneMaker's code. The best I can say is that I will and have passed such comments on. I will say it **is getting worked on** because we feel the same way about it - its out dated and needs an update.

Once again, and hopefully for the last time, thank you for your vote and I will pass it on. But I can't do anything else, despite my desire to see it improve too.

**Comment by tngreene** [comment]
Perhaps one thing that may make your more happy about fleets of legacy aircraft: A new secret project is being spec'd out to solve that for all artists - not just Blender 2.49 users. More details can be expected some time after v4.1.0-rc.1! Beyond that sentence, I won't say anymore. :)

**Comment by tngreene** [comment]
Now, I've got to leave this for a bit. Future bugs will break off from this to talk about forming these new habits and things to get organized talked about above.

**Comment by DWmFrancis** [comment]
Ted -

You mentioned passing my comments on to the right people.
Is it possible to communicate with them directly?
Is it Jennifer?

-df

**Comment by tngreene** [comment]
http://feedback.x-plane.com/ is the exact right place to go and no other! Jennifer is absolutely not the right choice.
